internal/testtypes/builders.go

```diff
@@ -0,0 +1,525 @@
+func (m *MachineDeploymentClassBuilder) WithBootstrapTemplate(t *unstructured.Unstructured) *MachineDeploymentClassBuilder {
+	m.bootstrapTemplate = t
+	return m
+}
+
+// WithClass sets the MachineDeploymentClass name for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithClass(class string) *MachineDeploymentClassBuilder {
+	m.class = class
+	return m
+}
+
+// WithReplicas sets the number of replicas for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithReplicas(replicas int32) *MachineDeploymentClassBuilder {
+	m.replicas = replicas
+	return m
+}
+
+// WithLabels sets the labels for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithLabels(labels map[string]string) *MachineDeploymentClassBuilder {
+	m.labels = labels
+	return m
+}
+
+// WithAnnotations sets the annotations for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithAnnotations(annotations map[string]string) *MachineDeploymentClassBuilder {
+	m.annotations = annotations
+	return m
+}
+
+// Build creates a full MachineDeploymentClass object with the variables passed to the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) Build() *clusterv1.MachineDeploymentClass {
+	return &clusterv1.MachineDeploymentClass{
+		Class: m.class,
+		Template: clusterv1.MachineDeploymentClassTemplate{
+			Metadata: clusterv1.ObjectMeta{
+				Labels:      m.labels,
+				Annotations: m.annotations,
+			},
+			Bootstrap: clusterv1.LocalObjectTemplate{
+				Ref: objToRef(m.bootstrapTemplate),
+			},
+			Infrastructure: clusterv1.LocalObjectTemplate{
+				Ref: objToRef(m.infrastructureMachineTemplate),
+			},
+		},
+	}
+}
+
+// InfrastructureMachineTemplateBuilder holds the variables and objects needed to build an InfrastructureMachineTemplate.
+type InfrastructureMachineTemplateBuilder struct {
+	namespace string
+	name      string
+}
+
+// NewInfrastructureMachineTemplateBuilder creates an InfrastructureMachineTemplateBuilder with the given name and namespace.
+func NewInfrastructureMachineTemplateBuilder(namespace, name string) *InfrastructureMachineTemplateBuilder {
+	return &InfrastructureMachineTemplateBuilder{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+// Build takes the objects and variables in the  InfrastructureMachineTemplateBuilder and generates an unstructured object.
+func (i *InfrastructureMachineTemplateBuilder) Build() *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(InfrastructureGroupVersion.String())
+	obj.SetKind(GenericInfrastructureMachineKind)
+	obj.SetNamespace(i.namespace)
+	obj.SetName(i.name)
+
+	if err := unstructured.SetNestedField(obj.UnstructuredContent(), true, "spec", "template", "spec", "fakeSetting"); err != nil {
+		panic(err)
+	}
+	return obj
+}
+
+// BootstrapTemplateBuilder holds the variables needed to build a generic template machineDeployment.spec.template.bootstrap.configRef.
```

Comment:
```suggestion
// BootstrapTemplateBuilder holds the variables and objects needed to build a generic BootstrapTemplate.
```
nit. I think that would be better as it doesn't directly refer to one of the use cases of BootstrapTemplates. (would be similar now to InfrastructureMachineTemplateBuilder)